api/pkg/service/rating/rating.go

```diff
@@ -59,26 +60,91 @@ func (s *service) Get(ctx context.Context, p *rating.GetPayload) (*rating.GetRes
 		return nil, err
 	}
 
-	var res model.Resource
-	if err := s.db.First(&res, p.ID).Error; err != nil {
-		if gorm.IsRecordNotFoundError(err) {
-			return nil, notFoundError
-		}
-		s.logger.Error(err)
-		return nil, fetchError
+	_, err = s.checkIfResourceExists(p.ID)
+	if err != nil {
+		return nil, err
 	}
 
-	q := s.db.Where("user_id = ? AND resource_id = ?", user.ID, p.ID)
+	q := s.db.Where(&model.UserResourceRating{UserID: user.ID, ResourceID: p.ID})
 
 	r := model.UserResourceRating{}
 	if err := q.Find(&r).Error; err != nil {
 		if gorm.IsRecordNotFoundError(err) {
 			return &rating.GetResult{Rating: -1}, nil
 		}
-
 		s.logger.Error(err)
 		return nil, fetchError
 	}
 
 	return &rating.GetResult{Rating: int(r.Rating)}, nil
 }
+
+// Update user's rating for a resource
+func (s *service) Update(ctx context.Context, p *rating.UpdatePayload) error {
+
+	user, err := s.Validator.UserFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	res, err := s.checkIfResourceExists(p.ID)
+	if err != nil {
+		return err
+	}
+
+	if err := s.updateUserRating(user.ID, p.ID, p.Rating); err != nil {
+		return err
+	}
+
+	return s.updateResourceRating(res, p.ID)
+}
+
+func (s *service) updateUserRating(userID, resourceID, rating uint) error {
+
+	q := s.db.Where(&model.UserResourceRating{UserID: userID, ResourceID: resourceID})
+
+	rat := &model.UserResourceRating{}
```

Comment:
`rat` is a bit amusing and confusing :D .. how about calling it `r` or `rating` 😂